_source/_posts/2022-06-01-jhipster-digital-ocean.md

```diff
@@ -0,0 +1,433 @@
+```
+
+Navigate to `http://localhost:8761` and sign in with your Okta credentials. When the registry shows all services in green, Set up port-forwarding for the gateway as well.
+
+```bash
+$ kubectl port-forward svc/gateway -n demo 8080
+```
+
+Navigate to `http://localhost:8080`, sign in, and create some entities to verify everything is working fine.
+
+After looking around, stop minikube before the cloud deployment:
+
+```bash
+$ minikube stop
+```
+
+## Deploy to DigitalOcean Kubernetes
+
+Now that the architecture works locally, let's proceed to the cloud deployment. First, create a [DigitalOcean](https://www.digitalocean.com/try/free-trial-offer) account, you can try their services with a free $100 credit that is available for 60 days. The registration requires a $5 PayPal payment, or providing a credit card.
+
+Most of the cluster tasks, if not all, can be accomplished using [doctl](https://github.com/digitalocean/doctl#installing-doctl), the command-line interface (CLI) for the DigitalOcean API. Install the tool, and perform the authentication with DigitalOcean:
+
+```bash
+$ doctl auth init
+```
+You will be prompted to enter the DigitalOcean access token that you can generate in the DigitalOcean control panel. Sign in, and then in the left menu go to **API**, click **Generate New Token**. Enter a token name, and click **Generate Token**. Copy the new token from the **Tokens/Keys** table.
+
+{% img blog/jhipster-digital-ocean/do-control-panel-token.png alt:"DigitalOcean Control Panel for Token Create" width:"500" %}{: .center-image }
+
+You can find a detailed list of pricing for cluster resources at [DigitalOcean](https://docs.digitalocean.com/products/kubernetes/),  and with `doctl` you can quickly retrieve a list of node size options available for your account:
+
+```bash
+$ doctl k options sizes
+```
+
+**NOTE**: I tested the cluster with the higher size Intel nodes available for my account, `s-2vcpu-4gb-intel`, in an attempt to run the application in the default cluster configuration, a three-node cluster with a single node pool in the nyc1 region, using the latest Kubernetes version. As I started to see pods not running due to "insufficient CPU", I increased the number of nodes later. DigitalOcean's latest and default Kubernetes version at the moment of writing this post is 1.22.8.
+
+Create the cluster with the following command line:
+
+```bash
+$ doctl k cluster create do1 -v --size s-2vcpu-4gb-intel
+```
+After creating a cluster, `doctl` adds a configuration context to kubectl and makes it active, so you can start monitoring your cluster right away with k9s. First, apply the resources configuration to the DigitalOcean cluster:
+
+```bash
+$ ./kubectl-apply.sh -f
+```
+
+Monitor the deployment with k9s:
+
+```bash
+$ k9s -n demo
+```
+
+{% img blog/jhipster-digital-ocean/k9s-do-cluster.png alt:"k9s user interface monitoring DigitalOcean Kubernetes" width:"800" %}{: .center-image }
```

Comment:
Thanks for updating the screenshot so I can compare the times with mine. I don't see a mongodb-2 pod and my mongodb-1 won't start.

<img width="1330" alt="Screen Shot 2022-05-20 at 15 48 02" src="https://user-images.githubusercontent.com/17892/169617008-9372b428-969e-4ee8-b616-9ae870284713.png">

It'd be nice to figure out a way to create a cluster and adjust it as needed so everything starts successfully the first time.

Reply:
As a test, I tried the following:

```
doctl k cluster create do1 -v --size s-4vcpu-8gb-intel
```

This seems to solve all my problems.

<img width="1442" alt="Screen Shot 2022-05-20 at 16 00 55" src="https://user-images.githubusercontent.com/17892/169618088-45764aa5-7de1-4657-af91-e0124826891d.png">

If you make this change, I think you can remove the Increase CPU and Increase Storage sections. Or, reword them to say, "when I first tried this with XXX size".

Reply:
I don't have that node size as an option. The biggest node I can use is s-2vcpu-4gb-intel. What kind of account do you have in DO? I will update the post to create the cluster with 5 nodes from the beginning, and increase cpu and storage will be a note.

Reply:
I changed the cluster create command. I had to submit a support ticket requesting higher node sizes for my account. I moved increase cpu and storage to a notes section.